integ-test/src/main/java/integtest/DoItNested.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integtest;
+
+import com.palantir.gradle.autoparallelizable.AutoParallelizable;
+import java.io.File;
+import java.util.stream.Collectors;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFile;
+
+@AutoParallelizable
+public final class DoItNested {
+    public abstract static class DoItNestedTask extends DoItNestedTaskImpl {
+        public DoItNestedTask() {
+            setDescription("lol");
+        }
+    }
+
+    public abstract static class AbstractNested {
+        @OutputFile
+        abstract RegularFileProperty getFileValue();
+    }
+
+    interface Nested {
+        @Input
+        Property<String> getStringValue();
+
+        @InputFiles
+        ConfigurableFileCollection getFilesValue();
+    }
+
+    interface DoubleNested {
+
+        @Input
+        SetProperty<Integer> getIntsValue();
+
+        @org.gradle.api.tasks.Nested
+        Nested getNested();
```

Comment:
Actually suprised this works, I thought all work params had to be a `Property`. I guess it actually everything has be able to be made withe `ObjectsFactory::newInstance`?

Reply:
everything [here](https://docs.gradle.org/current/userguide/custom_gradle_types.html#configuration_using_properties) I'm guessing